src/ast/passes/semantic_analyser.cpp

```diff
@@ -2824,35 +2834,59 @@ void SemanticAnalyser::visit(ExprStatement &expr)
   Visit(expr.expr);
 }
 
+static const std::unordered_map<Type, std::string_view> AGGREGATE_USAGES{
+  { Type::count_t, "count()" },
+  { Type::sum_t, "sum(retval)" },
+  { Type::min_t, "min(retval)" },
+  { Type::max_t, "max(retval)" },
+  { Type::avg_t, "avg(retval)" },
+  { Type::hist_t, "hist(retval)" },
+  { Type::lhist_t, "lhist(rand %10, 0, 10, 1)" },
+  { Type::stats_t, "stats(arg2)" },
+};
+
 void SemanticAnalyser::visit(AssignMapStatement &assignment)
 {
   Visit(assignment.map);
   Visit(assignment.expr);
 
-  auto type = assignment.expr->type;
-
-  if (!IsValidAssignment(type, assignment.expr->is_map)) {
-    auto hint = "";
-    if (type.IsHistTy()) {
-      hint = "hist(retval)";
-    } else if (type.IsLhistTy()) {
-      hint = "lhist(rand %10, 0, 10, 1)";
-    } else if (type.IsStatsTy()) {
-      hint = "stats(arg2)";
-    } else {
+  const auto *map_type = get_map_type(*assignment.map);
+  const bool is_map_declared = map_type != nullptr;
+  if (!IsValidAssignment(is_map_declared,
+                         assignment.expr->type,
+                         assignment.expr->is_map)) {
+    const auto &type = assignment.expr->type;
+    auto hint_usage = AGGREGATE_USAGES.find(type.GetTy());
+    if (hint_usage == AGGREGATE_USAGES.end())
       LOG(BUG) << "Missing assignment hint";
-    }
+
+    std::string_view hint_cast;
+    if (type.IsCastableMapTy())
+      hint_cast =
+          "\nAdd a cast to integer if you want the value of the aggregate.";
+
     LOG(ERROR, assignment.loc, err_)
         << "Map value '" << type
-        << "' cannot be assigned from one map to another. The function that "
-           "returns this "
-           "type must be called directly e.g. `@a = "
-        << hint << ";`.";
-  }
+        << "' cannot be assigned from one map to another. "
+           "The function that returns this type must be called directly e.g. `"
+        << assignment.map->ident << " = " << hint_usage->second << ";`."
+        << hint_cast;
+  }
+
+  // Add an implicit cast when copying the value of an aggregate map to an
+  // existing map of int. Enables the following: `@x = 1; @y = count(); @x = @y`
+  const bool map_contains_int = map_type != nullptr && map_type->IsIntTy();
+  const bool expr_is_map_with_castable_agg =
```

Comment:
Do we need `assignment.expr->is_map` ?

Reply:
Yes, otherwise it would allow the following:
```
@x = 0;         // Init @x with 0, so map_contains_int == true
@x = count();   // expr.type.IsCastableMapTy() == true, but expr.is_map == false
```